BUG: minimum versions CI is failing

### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.


### Reproducible Example

```python
pytest pandas/tests/io/formats/style/test_matplotlib.py::test_function_gradient
```


### Issue Description

The  following tests fail:
```
2022-11-03T09:37:13.6183475Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_function_gradient[background_gradient] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6184265Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_function_gradient[text_gradient] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6185065Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_color[background_gradient] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6185879Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_color[text_gradient] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6186701Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_axis[background_gradient-0-expected0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6187545Z FAILED pandas/tests/io/formats/style/test_style.py::TestStyler::test_subset_for_boolean_cols[background_gradient] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6188355Z FAILED pandas/tests/io/formats/style/test_style.py::TestStyler::test_subset_for_boolean_cols[text_gradient] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6189185Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_axis[background_gradient-1-expected1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6190053Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_axis[background_gradient-None-expected2] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6190896Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_axis[text_gradient-0-expected0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6191822Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_axis[text_gradient-1-expected1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6192669Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_axis[text_gradient-None-expected2] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6193488Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_text_color_threshold[PuBu-expected0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6194281Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_text_color_threshold[YlOrRd-expected1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6195093Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_vmin_vmax - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6195838Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_int64 - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6196629Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_array[0-gmap0-expected0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6197451Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_array[1-gmap1-expected1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6198271Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_array[None-gmap2-expected2] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6199136Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[None-exp_gmap0-gmap0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6199997Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[None-exp_gmap0-gmap1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6200860Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[None-exp_gmap0-gmap2] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6201707Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[None-exp_gmap0-gmap3] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6202553Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[None-exp_gmap0-gmap4] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6203420Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset1-exp_gmap1-gmap0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6204300Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset1-exp_gmap1-gmap1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6205160Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset1-exp_gmap1-gmap2] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6206026Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset1-exp_gmap1-gmap3] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6206874Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset1-exp_gmap1-gmap4] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6207802Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset2-exp_gmap2-gmap0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6208665Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset2-exp_gmap2-gmap1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6209524Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset2-exp_gmap2-gmap2] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6210438Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset2-exp_gmap2-gmap3] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6211289Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset2-exp_gmap2-gmap4] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6212154Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset3-exp_gmap3-gmap0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6213020Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset3-exp_gmap3-gmap1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6213886Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset3-exp_gmap3-gmap2] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6214744Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset3-exp_gmap3-gmap3] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6215997Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset3-exp_gmap3-gmap4] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6216876Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset4-exp_gmap4-gmap0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6217734Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset4-exp_gmap4-gmap1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6218603Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset4-exp_gmap4-gmap2] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6219471Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset4-exp_gmap4-gmap3] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6220318Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_dataframe_align[subset4-exp_gmap4-gmap4] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6221164Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_series_align[gmap0-0-exp_gmap0] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6221997Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_series_align[gmap1-1-exp_gmap1] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6222911Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_series_align[gmap2-0-exp_gmap2] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6223834Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_background_gradient_gmap_series_align[gmap3-1-exp_gmap3] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6224615Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_pass_colormap_instance[scatter] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6225373Z FAILED pandas/tests/io/formats/style/test_matplotlib.py::test_pass_colormap_instance[hexbin] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6226206Z FAILED pandas/tests/plotting/frame/test_frame_color.py::TestDataFrameColor::test_if_hexbin_xaxis_label_is_visible - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6227139Z FAILED pandas/tests/plotting/frame/test_frame_color.py::TestDataFrameColor::test_scatter_with_c_column_name_with_colors[c-Greys] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6228031Z FAILED pandas/tests/plotting/frame/test_frame_color.py::TestDataFrameColor::test_scatter_with_c_column_name_with_colors[color-Greys] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6228862Z FAILED pandas/tests/plotting/frame/test_frame_color.py::TestDataFrameColor::test_scatter_colorbar_different_cmap - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6229643Z FAILED pandas/tests/plotting/frame/test_frame_color.py::TestDataFrameColor::test_no_color_bar - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6230419Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_plot_scatter_with_c - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6231181Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_kind_both_ways - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6231939Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_hexbin_basic - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6232675Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_hexbin_with_c - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6233444Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_hexbin_cmap[kwargs0-BuGn] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6234259Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_hexbin_cmap[kwargs1-cubehelix] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6235065Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_hexbin_cmap[kwargs2-YlGn] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6235845Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_scatter_unknown_colormap - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6236602Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_memory_leak - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6237356Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_df_grid_settings - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6238104Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_plain_axes - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6238937Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_xlabel_ylabel_dataframe_plane_plot[hexbin-None-None] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6239883Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_xlabel_ylabel_dataframe_plane_plot[hexbin-X Label-None] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6240783Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_xlabel_ylabel_dataframe_plane_plot[hexbin-None-Y Label] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
2022-11-03T09:37:13.6241684Z FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_xlabel_ylabel_dataframe_plane_plot[hexbin-X Label-Y Label] - AttributeError: 'ColormapRegistry' object has no attribute 'get_cmap'
```

### Expected Behavior

I think they started failing since https://github.com/pandas-dev/pandas/pull/49412 cc @jbrockmendel @mroeschke 

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

--- a/ci/deps/actions-38-minimum_versions.yaml
+++ b/ci/deps/actions-38-minimum_versions.yaml
@@ -32,7 +32,7 @@ dependencies:
   - gcsfs=2021.07.0
   - jinja2=3.0.0
   - lxml=4.6.3
-  - matplotlib=3.6.0
+  - matplotlib=3.6.1
   - numba=0.53.1
   - numexpr=2.7.3
   - odfpy=1.4.1

--- a/doc/source/getting_started/install.rst
+++ b/doc/source/getting_started/install.rst
@@ -310,7 +310,7 @@ Can be managed as optional_extra with ``pandas[plot, output_formatting]``, depen
 ========================= ================== ================== =============================================================
 Dependency                Minimum Version    optional_extra     Notes
 ========================= ================== ================== =============================================================
-matplotlib                3.6.0              plot               Plotting library
+matplotlib                3.6.1              plot               Plotting library
 Jinja2                    3.0.0              output_formatting  Conditional formatting with DataFrame.style
 tabulate                  0.8.9              output_formatting  Printing in Markdown-friendly format (see `tabulate`_)
 ========================= ================== ================== =============================================================

--- a/doc/source/whatsnew/v2.0.0.rst
+++ b/doc/source/whatsnew/v2.0.0.rst
@@ -124,7 +124,7 @@ Optional libraries below the lowest tested version may still work, but are not c
 +=================+=================+=========+
 | pyarrow         | 6.0.0           |    X    |
 +-----------------+-----------------+---------+
-| matplotlib      | 3.6.0           |    X    |
+| matplotlib      | 3.6.1           |    X    |
 +-----------------+-----------------+---------+
 | fastparquet     | 0.6.3           |    X    |
 +-----------------+-----------------+---------+

--- a/pandas/compat/_optional.py
+++ b/pandas/compat/_optional.py
@@ -23,7 +23,7 @@ VERSIONS = {
     "gcsfs": "2021.07.0",
     "jinja2": "3.0.0",
     "lxml.etree": "4.6.3",
-    "matplotlib": "3.6.0",
+    "matplotlib": "3.6.1",
     "numba": "0.53.1",
     "numexpr": "2.7.3",
     "odfpy": "1.4.1",

